crates/circuit/src/circuit_instruction.rs

```diff
@@ -30,16 +30,21 @@ use crate::operations::{
 };
 use crate::packed_instruction::PackedOperation;
 
-/// These are extra mutable attributes for a circuit instruction's state. In general we don't
-/// typically deal with this in rust space and the majority of the time they're not used in Python
-/// space either. To save memory these are put in a separate struct and are stored inside a
-/// `Box` on `CircuitInstruction` and `PackedInstruction`.
 #[derive(Debug, Clone)]
+struct ExtraAttrs {
+    label: Option<String>,
+    duration: Option<PyObject>,
+    unit: Option<String>,
+    condition: Option<PyObject>,
+}
+
+/// Extra mutable attributes for a circuit instruction's state. In general we don't
+/// typically deal with this in rust space and the majority of the time they're not used in Python
+/// space either. To save memory, the attributes are stored inside a `Box` internally, so this
+/// struct is no larger than that.
+#[derive(Default, Debug, Clone)]
 pub struct ExtraInstructionAttributes {
-    pub label: Option<String>,
-    pub duration: Option<PyObject>,
-    pub unit: Option<String>,
-    pub condition: Option<PyObject>,
+    attributes: Option<Box<ExtraAttrs>>,
 }
```

Comment:
Any particular reason to use a record struct rather than a tuple struct?

Reply:
Nope! I thought of that later, but I think it makes sense.

Done in bca33aa9443b0122a8411e057e40f63b7d5eb59f.